### DISCUSSION TITLE
Nextjs 15 Nixpacks deploy fails

### DISCUSSION THREAD

**Discussion by meltinbit** [coollabsio/coolify#4632] [Q&A]
Hi, i tried deploy Nextjs 15  through Nixpacks and deploy fails cause **legacy peer deps** during the **npm install**.
Is there a way to force **npm install --legacy-peer-deps**?
Thanks.F.

**Comment by pascal13ch** [comment]
Hi @meltinbit 

Maybe you have already solved it, I had the same problem and solved it as follows:

You can create a `nixpacks.toml` configuration in the root directory and then define the steps (install, build) that should be executed:

With this configuration it worked for me:

```
[phases.setup]
providers = ["nodejs_18", "npm-9_x", "openssl", "curl", "wget"]

[phases.install]
cmds = ["npm ci --legacy-peer-deps"]

[phases.build]
cmds = ["npm run build"]

[start]
cmd = "npm run start"
```

**Comment by cairocoder** [comment]
Use Dockerfile option and create a new Dockerfile with below content and put it in the root directory of the project:

```
FROM node:20-alpine AS deps
WORKDIR /app

COPY package*.json ./
RUN npm ci --legacy-peer-deps

FROM node:20-alpine AS builder
WORKDIR /app
COPY --from=deps /app/node_modules ./node_modules
COPY . .
ENV NEXT_TELEMETRY_DISABLED=1
RUN npm run build

FROM node:20-alpine AS runner
WORKDIR /app
ENV NODE_ENV=production
ENV NEXT_TELEMETRY_DISABLED=1

COPY --from=builder /app/package*.json ./
COPY --from=builder /app/.next ./.next
COPY --from=builder /app/public ./public
COPY --from=builder /app/node_modules ./node_modules
COPY --from=builder /app/next.config.* ./
COPY --from=builder /app/package.json ./package.json

EXPOSE 3000
CMD ["npm", "start"]
```
